pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -269,9 +270,11 @@ public void run() {
 
                 try {
                     processResult(currentRecord, result);
-                } catch (Exception e) {
+                    result.join();
+                } catch (CompletionException e) {
                     log.warn("Failed to process result of message {}", currentRecord, e);
                     currentRecord.fail();
+                    throw e.getCause();
```

Comment:
Re-throw the real exception

Reply:
please also handle InterruptedException as well